generic_camera_model_sampler.py

```diff
@@ -0,0 +1,67 @@
+
+from colorama import Fore, Style
+
+from .planar_as_base import ( PlanarAsBase, INTER_MAP, input_2_torch, torch_2_output )
+from .camera_model_sampler import CameraModelRotation
+from .register import (SAMPLERS, register)
+from ..mvs_utils.ftensor import FTensor
+
+@register(SAMPLERS)
+class GenericCameraModelSampler(CameraModelRotation):
+    def __init__(self, camera_model_raw, camera_model_target, R_output_in_input, preprocessing = lambda x: x, postprocessing = lambda x: x):
+        '''
+        This is a wrapper class for the CameraModelRotation class.
+        It generalizes the operation of the sampler to also take in a preprocessing function and a postprocessing function.
+
+        The raw image is a planer image that described by a camera model. 
+        We create the target image by preprocessing the input image, sampling from the preprocessed image (may no longer be an image), and postprocessing the result.
+
+        R_output_in_input is the rotation matrix measured in the raw (input) image frame. 
+        The coordinates of a 3D point in the target camera image frame x_f can 
+        be transformed to the point in the raw image frame x_p by
+        x_p = R_output_in_input @ x_f.
+
+        R_output_in_input is following the naming convention. This means that CIF's orientation
```

Comment:
This is lack context. I left this doc string in the wrong state in the first place. I will fix it later after the merge.

Reply:
:parrot: :+1: